rust/cmsis-update/src/download.rs

```diff
@@ -84,89 +116,173 @@ impl DownloadProgress for () {
     }
 }
 
-pub struct DownloadContext<'a, Conf, Prog, Con>
-where Conf: DownloadConfig,
-      Prog: DownloadProgress + 'a,
-      Con: Connect,
+pub struct DownloadContext<'a, Conf, Prog>
+where
+    Conf: DownloadConfig,
+    Prog: DownloadProgress + 'a,
 {
     config: &'a Conf,
-    prog:  Prog,
-    client: &'a Client<Con, Body>,
+    prog: Prog,
+    client: Client,
     log: &'a Logger,
 }
 
-impl<'a, Conf, Prog, Con> DownloadContext<'a, Conf, Prog, Con>
-where Conf: DownloadConfig,
-      Prog: DownloadProgress + 'a,
-      Con: Connect,
+impl<'a, Conf, Prog> DownloadContext<'a, Conf, Prog>
+where
+    Conf: DownloadConfig,
+    Prog: DownloadProgress + 'a,
 {
-    pub fn new(config: &'a Conf, prog: Prog, client: &'a Client<Con, Body>, log: &'a Logger) -> Self {
-        DownloadContext {
+    pub fn new(config: &'a Conf, prog: Prog, log: &'a Logger) -> Result<Self, Error> {
+        let client = ClientBuilder::new()
+            .use_rustls_tls()
```

Comment:
@JanneKiiskila Removing that line forces linux users to use Openssl instead of Rustls. This prevents the package from being a "manylinux" wheel, and you're only allowed to upload "manylinux" wheels to pypi. My hands are tied, I can't use Openssl here.

Reply:
Oh, that is a problem. :-/ Question is then can we file an issue to the RustLS and have those guys fixed the underlying TLS handshake/cipher issue?

Reply:
That really depends on the cypher suite is question. They have no plans to add support for known-broken cyphers.

Reply:
KEIL or AWS should not use broken cipher suites. The chosen cipher suite is selected with a negotiation, right? If the client proposes a good, working cipher suite - surely the back-end should use it?

Reply:
I agree, they should not use broken cypher suites. also, I think the hosting is Microsoft related, at least that's what I recall. It's also really frustrating to debug this issue, as the server in question simply terminates the connection instead of giving us any information. We might have negotiated for a cypher suite that segfaults for all we know.

Reply:
Seems we're getting more failures... :-( 

```
(cmsis) jankii01@ubuntu:~/mbed/mbed-bootloader/mbed-os/tools$ python project.py --update-packs
Nov 12 16:04:06.756 ERRO download of "http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC5528_DFP.pdsc" failed: http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC5528_DFP.pdsc: connection closed before message completed
Nov 12 16:04:06.784 ERRO download of "http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC5526_DFP.pdsc" failed: http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC5526_DFP.pdsc: connection closed before message completed
Nov 12 16:04:06.807 ERRO download of "http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC55S28_DFP.pdsc" failed: http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC55S28_DFP.pdsc: connection closed before message completed
Nov 12 16:04:06.996 ERRO download of "http://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC55S26_DFP.pdsc" failed: https://mcuxpresso.nxp.com/cmsis_pack/repo/NXP.LPC55S26_DFP.pdsc: connection closed before message completed
Nov 12 16:04:20.926 ERRO download of "http://www.amiccom.com.tw/download/Amiccom_DeviceFamilyPack/Amiccom.SoC_DFP.pdsc" failed: http://www.amiccom.com.tw/download/Amiccom_DeviceFamilyPack/Amiccom.SoC_DFP.pdsc: error trying to connect: Connection refused (os error 111)
```